baseapp/baseapp.go

```diff
@@ -688,6 +689,17 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte) (gInfo sdk.GasInfo, re
 	// Result if any single message fails or does not have a registered Handler.
 	result, err = app.runMsgs(runMsgCtx, msgs, mode)
 	if err == nil && mode == runTxModeDeliver {
+		// Run optional postHandlers
+		if app.postHandler != nil {
+			newCtx, err := app.postHandler(runMsgCtx, tx, mode == runTxModeSimulate)
+
```

Comment:
```suggestion
```

Reply:
So to be clear, if the post-handler fails, we don't write ANY of the message execution's state?

Reply:
Yes, I added a comment.

Rationale is to mimic how tips were implemented in the previous middlewares:
- in antehandlers, deduct fees from feepayer
- run msgs
- if msgs are succesful, transfer tip to feepayer
- if that transfer is not successful (e.g. if tipper doesn't have enough tip balance), then revert the msg, because it's the tipper's fault, so his msgs shouldn't be executed.

Reply:
However, we could decide to do 3 store branches: one ante, one runMsgs, one post. What's your opinion on that @alexanderbez?

If we do that, the tip posthandler would need to be rewritten.

Reply:
@amaurym, this is how I see it:

1. IFF you treat `runMsgs` and `postHandlers` as **_atomic_**, then they should use the same branch.
2. If, on the other hand, they are NOT atomic, i.e. `runMsgs` passes and `postHandlers` fails but still commit, then they should use different branches.

Being that you've implemented it as atomic, i.e. (1), then it makes sense to use the same branch as you do now.